src/src/PluginStructs/P122_data_struct.cpp

```diff
@@ -0,0 +1,568 @@
+    {
+      String log = F("SHT2x : ***state transition ");
+      log += String((int)oldState);
+      log += F("-->");
+      log += String((int)_state);
+      addLog(LOG_LEVEL_INFO, log);
+    }
+  }
+  return stable;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Returns the I2C connection state
+// Note: based upon the FSM state without actual accessing the device
+bool P122_data_struct::isConnected()
+{
+  switch (_state)
+  {
+    case  P122_state::Initialized:
+    case  P122_state::Ready:
+    case  P122_state::Wait_for_temperature_samples:
+    case  P122_state::Wait_for_humidity_samples:
+    case  P122_state::New_Values_Available:
+      return true;
+      break;
+    case  P122_state::Uninitialized:
+    case  P122_state::Error:
+    case  P122_state::Wait_for_reset:
+    default:
+      return false;
+      break;
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Returns if the device communication is in error
+// Note: based upon the FSM state without actual accessing the device
+bool P122_data_struct::inError()
+{
+  return _state == P122_state::Error;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Returns if new acquired values are available
+bool P122_data_struct::newValues()
+{
+  return _state == P122_state::New_Values_Available;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Restart the FSM used to access the device
+bool P122_data_struct::reset()
+{
+  _state = P122_state::Uninitialized;
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Start a new measurement cycle
+bool P122_data_struct::startMeasurements()
+{
+  if ((_state == P122_state::New_Values_Available) || (_state == P122_state::Initialized))
+  {
+    _state = P122_state::Ready;
+  }
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Get the electronic idenfification data store in the device
+// Note: The data is read from the device during initialization 
+bool P122_data_struct::getEID(uint32_t *eida, uint32_t *eidb, uint8_t *firmware)
+{
+  *eida = _eida;
+  *eidb = _eidb;
+  *firmware = _firmware;
+  return true;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Return the previously measured temperature in [C]
+float P122_data_struct::getTemperature()
+{
+  return -46.85 + (175.72 / 65536.0) * _rawTemperature;  // Datasheet par 6.2
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
+// Return the previously measured humidity in [%RH]
+float P122_data_struct::getHumidity()
+{ 
+  return -6.0 + (125.0 / 65536.0) * _rawHumidity;  // Datasheet par  6.1
```

Comment:
Please append `f` to those floating point "magic values".
Thus `6.0` -> `6.0f`, etc.

Otherwise the compiler will consider them to be `double` instead of `float`